pynemo/tide/nemo_bdy_tide3.py

```diff
@@ -1,272 +1,291 @@
+    sinz = np.zeros((numharm, ampz.shape[1]))
+    cosuX = np.zeros((numharm, nbdyu))
+    sinuX = np.zeros((numharm, nbdyu))
+    cosvX = np.zeros((numharm, nbdyu))
+    sinvX = np.zeros((numharm, nbdyu))
+    cosuY = np.zeros((numharm, nbdyv))
+    sinuY = np.zeros((numharm, nbdyv))
+    cosvY = np.zeros((numharm, nbdyv))
+    sinvY = np.zeros((numharm, nbdyv))
+
+    compindx = constituents_index(tpxo_z.cons, comp)
+    for h in range(0, numharm):
+        c = int(compindx[h])
+        if c != -1:
+            cosz[h, :] = ampz[c, :] * np.cos(np.deg2rad(phaz[c, :]))
+            sinz[h, :] = ampz[c, :] * np.sin(np.deg2rad(phaz[c, :]))
+
+            if key_transport == 1:
+                if (np.sum(depu[:] <= 0.0) > 0) | (np.sum(depv[:] <= 0.0) > 0):
+                    logger.error(' Error: Land or Mask contamination')
+
+                cosuX[h, :] = ampuX[c, :] * np.cos(np.deg2rad(phauX[c, :])) / depu
+                sinuX[h, :] = ampuX[c, :] * np.sin(np.deg2rad(phauX[c, :])) / depu
+                cosvX[h, :] = ampvX[c, :] * np.cos(np.deg2rad(phavX[c, :])) / depu
+                sinvX[h, :] = ampvX[c, :] * np.sin(np.deg2rad(phavX[c, :])) / depu
+                cosuY[h, :] = ampuY[c, :] * np.cos(np.deg2rad(phauY[c, :])) / depv
+                sinuY[h, :] = ampuY[c, :] * np.sin(np.deg2rad(phauY[c, :])) / depv
+                cosvY[h, :] = ampvY[c, :] * np.cos(np.deg2rad(phavY[c, :])) / depv
+                sinvY[h, :] = ampvY[c, :] * np.sin(np.deg2rad(phavY[c, :])) / depv
+            else:
+                cosuX[h, :] = 0.01 * ampuX[c, :] * np.cos(np.deg2rad(phauX[c, :]))
+                sinuX[h, :] = 0.01 * ampuX[c, :] * np.sin(np.deg2rad(phauX[c, :]))
+                cosvX[h, :] = 0.01 * ampvX[c, :] * np.cos(np.deg2rad(phavX[c, :]))
+                sinvX[h, :] = 0.01 * ampvX[c, :] * np.sin(np.deg2rad(phavX[c, :]))
+                cosuY[h, :] = 0.01 * ampuY[c, :] * np.cos(np.deg2rad(phauY[c, :]))
+                sinuY[h, :] = 0.01 * ampuY[c, :] * np.sin(np.deg2rad(phauY[c, :]))
+                cosvY[h, :] = 0.01 * ampvY[c, :] * np.cos(np.deg2rad(phavY[c, :]))
+                sinvY[h, :] = 0.01 * ampvY[c, :] * np.sin(np.deg2rad(phavY[c, :]))
+
+# TOD:: Do we need to rotate ??? And is this method  correct ????
+    maxJ = DC.lonlat['t']['lon'].shape[0]
+    maxI = DC.lonlat['t']['lon'].shape[1]
+    dst_gcos = np.ones([maxJ, maxI])
+    dst_gsin = np.zeros([maxJ, maxI])
+    #lets start with the u-points
+    grid_angles = nemo_bdy_grid_angle.GridAngle(setup.settings['dst_hgr'], 0, maxI, 0, maxJ, 'u')
+    dst_gcos = grid_angles.cosval
+    dst_gsin = grid_angles.sinval
+
+    #retain only boundary points rotation information
+    tmp_gcos = np.zeros((Grid_U.bdy_r==0).sum())
+    tmp_gsin = np.zeros((Grid_U.bdy_r==0).sum())
+    for index in range(len(tmp_gcos)):
+        tmp_gcos[index] = dst_gcos[Grid_U.bdy_i[index, 1], Grid_U.bdy_i[index, 0]] # Iterate over bdy_i (not bdy_r )because the
```

Comment:
I assume that this is OK - to iterate over `bdy_i` for the first `bdy_r==0` elements?